src/mono/netcore/System.Private.CoreLib/src/System/Delegate.Mono.cs

```diff
@@ -424,15 +424,7 @@ private static bool IsArgumentTypeMatchWithThis(Type delArgType, Type argType, b
         {
             if (Method is null)
             {
-#nullable disable
-                // FIXME: This code cannot handle null argument values
-                Type[] mtypes = new Type[args.Length];
-                for (int i = 0; i < args.Length; ++i)
-                {
-                    mtypes[i] = args[i].GetType();
-                }
-                method_info = _target.GetType().GetMethod(data.method_name, mtypes);
-#nullable restore
+                    throw new NullReferenceException ("method_info is null");
```

Comment:
What's the right way to write something like this for the BCL in a way that is pleasing to the nullability checker?

Reply:
basically I want to do an explicit null check of `Method` early rather than at the place where it's used later in this method.

Reply:
this sounds like a good way to do it, are you saying the nullability checker can't figure out that `Method` is non-null later on?

Reply:
it can figure it out; I keep forgetting that properties are assumed not to change within a function.  In any case, I cleaned up my changes a bit